src/v/config/tests/enterprise_property_test.cc

```diff
@@ -96,6 +100,105 @@ TEST(EnterprisePropertyTest, TestRestriction) {
     EXPECT_TRUE(cfg.enterprise_enum.check_restricted(N(tls_version::v1_3)));
 }
 
+template<typename Property>
+struct EnterprisePropertyTest : public ::testing::Test {
+    using value_type = typename Property::underlying;
+
+    struct test_values {
+        value_type default_value;
+        value_type allowed_value;
+        value_type restricted_value;
+    };
+
+    struct test_case {
+        enterprise<Property>* enterprise_conf;
```

Comment:
nit: would this be simpler as a reference?
```suggestion
        enterprise<Property>& enterprise_conf;
```

Reply:
yes, it would. But clang-tidy is complaining that you have a non-copyable/non-movable struct.
The options is to either use `std::reference_wrapper` (or similar) or add a NOLINT directive. If find the option of just using a pointer (in a test), the least bad option among these 3.